src/main/kotlin/org/jetbrains/research/testspark/tools/llm/Llm.kt

```diff
@@ -35,11 +36,11 @@ class Llm(override val name: String = "LLM") : Tool {
     private fun getLLMProcessManager(project: Project, psiFile: PsiFile, caretOffset: Int, testSamplesCode: String): LLMProcessManager {
         val classesToTest = mutableListOf<PsiClass>()
 
-        ApplicationManager.getApplication().runReadAction(
-            Computable {
+        ProgressManager.getInstance().runProcessWithProgressSynchronously({
+            ApplicationManager.getApplication().runReadAction {
```

Comment:
If it runs synchronously, does it block the main thread during the execution (as it was before for test compilation)?

Reply:
I think, that is the worst case, it wil lead to lackness of test samples. Not blocking the main thread.

Reply:
I added the sleep function call to test the blocking:
```kotlin
ApplicationManager.getApplication().runReadAction {
    collectClassesToTest(project, classesToTest, psiFile, caretOffset)
    sleepCancellable(20000)
}
// if sleepCancellable(20000) is placed outside the runReadAction but within the runProcessWithProgressSynchronously, it still blocks
```
It blocks the entire UI; here on the image I cannot click anything within the IDE except for dragging the popup window:

<img width="1536" alt="Screenshot 2024-05-24 at 16 32 28" src="https://github.com/JetBrains-Research/TestSpark/assets/47076189/897348c3-3c43-460f-8d84-9118b25e554f">

It is generally better to have a non-freezing UI, but I have not yet found a solution to this. Maybe something like `ProgressManager.getInstance().runProcessWithProgressAsynchronously()` may help.

Reply:
thank you so much for testing it, yes, that's an incorrect soluiton from my side. I'll work on it